configure.ac

```diff
@@ -2080,6 +2080,34 @@ case $CC in
         fi
 esac
 
+# Historically, some of our code assumed that signed integer overflow
+# is defined behaviour via twos-complement.  Prepare some flags so that
+# we can mark the modules that still do so.
+_SAVE_VAR([CFLAGS])
+CFLAGS="-fstrict-overflow -fno-strict-overflow"
+AC_CACHE_CHECK([if $CC supports -fstrict-overflow and -fno-strict-overflow],
+               [ac_cv_cc_supports_fstrict_overflow],
+  AC_COMPILE_IFELSE(
+    [
+      AC_LANG_PROGRAM([[]], [[]])
+    ],[
+      STRICT_OVERFLOW_CFLAGS="-fstrict-overflow"
+      NO_STRICT_OVERFLOW_CFLAGS="-fno-strict-overflow"
+    ],[
+      STRICT_OVERFLOW_CFLAGS=""
+      NO_STRICT_OVERFLOW_CFLAGS=""
+    ])
+)
+_RESTORE_VAR([CFLAGS])
+
+AC_MSG_CHECKING(for --with-strict-overflow)
+AC_ARG_WITH(strict-overflow,
+  AS_HELP_STRING(
+    [--with-strict-overflow],
+    [choose between -fstrict-overflow or -fno-strict-overflow (default is no for backwards compatibility)]),,
+  with_fstrict_overflow=no)
+AC_MSG_RESULT($with_strict_overflow)
```

Comment:
autoconf-good-practice-nit: please wrap all macro arguments in braces :)

Also, I think brevity is better: _"default is no for backwards compatibility"_ => _"default is no"_